src/node/coverage.ts

```diff
@@ -0,0 +1,267 @@
+// Future goals:
+// - Persist owner/repo names in workspace storage
+// - Cache coverage
+// - Show coverage totals somewhere
+// - Maybe add codecov button to disable and/or view in codecov
+
+import {
+  ExtensionContext,
+  OverviewRulerLane,
+  Position,
+  Range,
+  Uri,
+  commands,
+  window,
+  workspace,
+} from "vscode";
+import axios from "axios";
+
+type Coverage =
+  | {
+      line_coverage: [number, number][];
+    }
+  | undefined;
+
+const Colors = {
+  covered: "rgb(33,181,119)",
+  partial: "rgb(244,176,27)",
+  missed: "rgb(245,32,32)",
+} as const;
+
+const Icons = {
+  covered: Uri.parse(
+    "data:image/svg+xml;base64," +
+      Buffer.from(
+        `<svg version="1.1" width="18" height="18" xmlns="http://www.w3.org/2000/svg">
+  <rect width="3" height="100%" fill="${Colors.covered}" />
+</svg>`
+      ).toString("base64")
+  ),
+  partial: Uri.parse(
+    "data:image/svg+xml;base64," +
+      Buffer.from(
+        `<svg version="1.1" width="18" height="18" xmlns="http://www.w3.org/2000/svg">
+  <rect width="3" height="100%" fill="${Colors.partial}" />
+</svg>`
+      ).toString("base64")
+  ),
+  missed: Uri.parse(
+    "data:image/svg+xml;base64," +
+      Buffer.from(
+        `<svg version="1.1" width="18" height="18" xmlns="http://www.w3.org/2000/svg">
+  <rect width="3" height="100%" fill="${Colors.missed}" />
+</svg>`
+      ).toString("base64")
+  ),
+} as const;
+
+export function activateCoverage(context: ExtensionContext) {
+  const command = "codecov.reset.api.key";
+
+  const resetApiKeyHandler = () => {
+    context.secrets.delete("api.key");
+    updateDecorations();
+    console.log("reset api key");
```

Comment:
do we want this? Maybe something a little clearer for the client to interpret what exactly happened, like "API Key has been successfully reset" idk